# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PWindowGlobal.ipdl
# Commit: 40b61344f0da
# Full Hash: 40b61344f0da14f9700a75a8a1c67458e440b68a
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-08-15 09:46:49
# Regressor Bug: 1562292
# File Overlap Count: 1
# Description:
#   Bug 1562292: Part 1f - Implement BrowsingContext::LoadURI. r=nika
#   
#   In order to do cross-process targeted window.open() and link click operations,
#   we need a way to load URIs in the current DocShell of a BrowsingContext,
#   whichever process it lives in.
# ==============================================================================

diff -r 3bffe760e2db -r 40b61344f0da dom/ipc/PWindowGlobal.ipdl
--- a/dom/ipc/PWindowGlobal.ipdl	Thu Aug 01 16:36:32 2019 -0700
+++ b/dom/ipc/PWindowGlobal.ipdl	Thu Aug 01 18:00:32 2019 -0700
@@ -4,6 +4,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+include "mozilla/dom/DocShellMessageUtils.h";
+
 include protocol PBrowser;
 include protocol PInProcess;
 include protocol PBrowserBridge;
@@ -14,6 +16,7 @@
 using mozilla::gfx::IntRect from "mozilla/gfx/Rect.h";
 using moveonly mozilla::gfx::PaintFragment from "mozilla/gfx/CrossProcessPaint.h";
 using nscolor from "nsColor.h";
+using refcounted class nsDocShellLoadState from "nsDocShellLoadState.h";
 
 namespace mozilla {
 namespace dom {
@@ -50,10 +53,17 @@
    */
   async GetSecurityInfo() returns(nsCString? serializedSecInfo);
 
+  async LoadURIInChild(nsDocShellLoadState aLoadState);
+
 both:
   async RawMessage(JSWindowActorMessageMeta aMetadata, ClonedMessageData aData);
 
 parent:
+  // Load the given URI load state into the current owner process of the given
+  // BrowsingContext. aTargetBC must be in the same BrowsingContextGroup as this
+  // window global.
+  async LoadURI(BrowsingContext aTargetBC, nsDocShellLoadState aLoadState);
+
   /// Update the URI of the document in this WindowGlobal.
   async UpdateDocumentURI(nsIURI aUri);
 